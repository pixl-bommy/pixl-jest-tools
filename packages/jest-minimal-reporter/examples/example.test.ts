// these test should show each type of test result

describe("two level block", () => {
    describe.each([0, 1, 2, 3, 4])("subblock %p", () => {
        test.each([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])("test %p succeeds", () => {
            return;
        });
    });
});

test("succeeding test", () => {
    return;
});

test("failing test", () => {
    throw "error";
});

test("pending test", async () => {
    return new Promise(() => {
        // nothing will be done
    });
});

test("undone test", (done) => {
    // nothing to be done
});

test.skip("skipped test", () => {
    // nothing will be done
});

test.todo("not implemented test");

describe("failing block", () => {
    describe("sub block in failing block", () => {
        test.skip("skipped test", () => {
            expect(null).toBeNull();
        });
    });

    describe.skip("skipped sub block in failing block", () => {
        test("should be skipped", () => {
            expect(null).toBeNull();
        });
    });
});

describe("snapshot tests", () => {
    test("successful snapshot test", () => {
        expect("Kekse").toMatchSnapshot();
    });

    test("failing snapshot test", () => {
        const failingSnapshot = "" + Math.floor(Math.random() * 1000000);
        expect(failingSnapshot).toMatchSnapshot();
    });
});
