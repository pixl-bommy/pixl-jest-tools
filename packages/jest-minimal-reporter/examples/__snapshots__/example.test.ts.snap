// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests failing snapshot test 1`] = `"80245"`;

exports[`snapshot tests obsolete snapshot test 1`] = `"obsolete"`;

exports[`snapshot tests successful snapshot test 1`] = `"Kekse"`;
